File: C++/Problems/Problems/problem_2178.cpp
#include "problem_2178.h"

bool problem_2178::test() {
    int finalSum = 12;

    std::vector<long long> expected  { 2, 4, 6 };

    auto result = maximumEvenSplit(finalSum);

    print1DVector(result);

    return expected == result;
}

std::vector<long long> problem_2178::maximumEvenSplit(long long finalSum) {
    long long num = (long long)((std::sqrt(4 * finalSum + 1) - 1) / 2);
    std::vector<long long> result(num);
    for (int i = 0; i < num; ++i)
        result[i] = 2 * i + 2;
    int bonus = finalSum - 2 * (num * (num + 1) / 2);
    if (bonus % 2 == 1)
        return std::vector<long long> {};
    result[num - 1] += bonus;
    return result;
}

File: Python/Problems/Problems/problem2178.py
import heapq
import math
import queue
import sys

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter, deque

from .problem import Problem

class Problem2178(Problem):
    def test(self) -> bool:
        finalSum = 12

        expected = [2, 4, 6]

        result = self.maximumEvenSplit(finalSum)

        print1DArray(result)

        return expected == result

    def maximumEvenSplit(self, finalSum: int) -> List[int]:
        num = int((math.sqrt(4 * finalSum + 1) - 1) / 2)
        result = []
        for i in range(num):
            result.append(2 * i + 2)
        bonus = finalSum - 2 * (num * (num + 1) // 2)
        if bonus % 2 == 1:
            return []
        result[num - 1] += bonus
        return result